Src/SemanticComparison/ProxyGenerator.cs

```diff
@@ -173,13 +183,10 @@ private static void BuildMethodGetHashCode<TDestination>(TypeBuilder type)
             method.SetReturnType(typeof(int));
 
             int derivedGetHashCode = 135;
-            MethodInfo getHashCode = typeof(TDestination).GetMethod(

```

Comment:
Well.. I've run all the unit tests in both .NET Framework and .NET Core platforms and they didn't fail. Could we trust tests in this project?

The only change is that now I'm looking for the public method rather than for public & private. Please review and let me know whether it does matter. From what I see, it doesn't matter as we select public methods only.